### DISCUSSION TITLE
The ItemTemplateSelector of ItemsControl is not being triggered on Windows

### DISCUSSION THREAD

**Discussion by WenHong-Lau** [unoplatform/uno#15371] [Q&A] [answered]
The ItemTemplateSelector is not being triggered on Windows, resulting in a failure to render controls as expected. This issue is specific to the following XAML:

```
<ItemsControl Grid.Row="2" ItemsSource="{Binding Persons, Mode=TwoWay}" ItemTemplateSelector="{StaticResource personTemplateSelector}"/>
``` 

 On Android, the ItemTemplateSelector functions as intended, correctly rendering controls based on the provided templates. This is how it look like on Android:
 
![image](https://github.com/unoplatform/uno/assets/88780667/1a6ab350-0cfa-4e0a-8f0b-d6ede6411fee)

On Windows, the ItemTemplateSelector not triggered, leading to controls not rendered. This is how it look like on Windows:
![image](https://github.com/unoplatform/uno/assets/88780667/2615f637-495e-4a89-9f12-c43adbe703bc)

**Comment by jeromelaban** [comment]
Thanks for the report. To be sure about what you're calling Windows, are you referring to the `.Windows` project?

**Reply by WenHong-Lau** [reply]
That's right, the .Windows project.

**Reply by jeromelaban** [reply]
Thanks. This would mean that Uno is doing something differently than WinUI, which is quite rare. In all cases, Uno is not present when running in the `.Windows` head, which may indicate a bug in WinUI. If you can [attach a repro](https://platform.uno/docs/articles/uno-howto-create-a-repro.html?tabs=windows-vs) of the issue here, it will help troubleshoot further.

**Reply by WenHong-Lau** [reply]
[UnoApp2.zip](https://github.com/unoplatform/uno/files/14207813/UnoApp2.zip)
Here's the zipped solution

**Reply by jeromelaban** [reply]
Thanks. This will likely be a report to make to the WinUI team as well.

**Comment by nickrandolph** [comment, marked as answer]
![image](https://github.com/unoplatform/uno/assets/1614057/e099ca54-c76c-4064-921c-175e9bde2c81)
The issue appears to be that the `SelectTemplateCore(object item)` overload doesn't get invoked by ItemsControl on Windows. However, if you put the logic in the `SelectTemplateCore(object item, DependencyObject container)` overload it does get called and the template selection works.
I'll close this issue as the solution is to use the alternative overload on the `DataTemplateSelector`
